### DISCUSSION TITLE
invalid BigNumber string ERROR

### DISCUSSION THREAD

**Discussion by MDimow** [hashlips-lab/nft-erc721-collection#444] [General]
Hello, everybody.

I am getting this problem when I try to set the _Cost before deploying when I use decimals.

**creation of YourNftToken errored: Error encoding arguments: Error: invalid BigNumber string (argument="value", value="0.1", code=INVALID_ARGUMENT, version=bignumber/5.5.0)**

On the other hand, if I use a whole number like 1 for the _Cost it accepts it and the contract is being deployed, BUT! After I try to mint it returns the same error:

**transact to GPTpederastiq.mint errored: Error encoding arguments: Error: invalid BigNumber string (argument="value", value="", code=INVALID_ARGUMENT, version=bignumber/5.5.0)**

P.S. I didn't change anything in the contract. I am using Remix with compiler 0.8.9

**Comment by liarco** [comment]
Duplicate of #445
